### ISSUE TITLE
Run the project on Zynq board

### ISSUE THREAD

**Issue by Adnan1986** [hakiri/openTSN#2]
Hello,
 I am a student and I am working on the implementation of TSN on FPGA and this work is very interesting for me.

Base on open source code materials, at first I have regenerated the hardware component (TSNSwitch and TSNNic) in my vivado. Now, I am going to compile the software component( it should be TSNLight) in my vivado SDK, but it sounds like a lot of files in source code and I have less experience in software. Could you please give me some advice about how I can compile the software part? till I use it beside the hardware component to have a complete TSN switch inside my zynq board.
Your help is really appreciated.
Many thanks in advance.

**Comment by hakiri** [comment]
Please use the documentation in the PDF files inside doc directory. Use translation if needed !

**Comment by Adnan1986** [comment]
I did. But the document is talked about how to run it on openbox. there is no explanation on how to run it on Vivado SDK.

**Comment by hakiri** [comment]
Look at openbox main idea in this paper:
https://conferences.sigcomm.org/sigcomm/2015/pdf/papers/hotmiddlebox/p67.pdf

**Comment by hakiri** [comment]
See also: https://dl.acm.org/doi/pdf/10.1145/3326285.3329067

**Comment by hakiri** [comment]
Vivado SDK is described here (use translation) : https://github.com/hakiri/openTSN/blob/master/sys/fast/fast_os/OpenTSN%E7%A1%AC%E4%BB%B6%E5%BC%80%E5%8F%91%E6%89%8B%E5%86%8C.pdf

**Comment by Adnan1986** [comment]
Many thanks for sharing, I will read them.

**Comment by hakiri** [comment]
@Adnan1986 all files are inside the sys/fast_os directory, you will find all PDF documents submitted with the original repos !
The original work was written in Chinese, you will need an automatic translation tool like G. translator.
The repo, including the materials and files included are only for learning purpose,

**Comment by Adnan1986** [comment]
Yes, there are a lot of files which I need to read all of them. Indeed I am a student and this project is kind of help me a lot if I could understand it all. I looked to a vivado part:(Vivado SDK is described here (use translation) : https://github.com/hakiri/openTSN/blob/master/sys/fast/fast_os/OpenTSN%E7%A1%AC%E4%BB%B6%E5%BC%80%E5%8F%91%E6%89%8B%E5%86%8C.pdf) as you said.
And it sounds like this part is just the implementation of TSNSwitch modules of OPENTSN for specific hardware and it is not cover the whole project implementation for the Zynq board.

As I read an OpenTSN paper, there are three main modules in this project (TSNNic, TSNSwitch for hardware, and TSNLight for software).
As I said I was trying to implement and run all three modules separately at first for my specific ZYNQ board on Vivado+SDK. then with a combination of all of them create OpenTSN on my Zynq board. For now, I figure out to synthesis of the Hardware part (TSNswitch and TSNNic). My next step is running software components in SDK. But there are a lot of files inside of software folder(./opentsn-openTSN-master\openTSN\src\软件代码) which I don't know how to run specific components that I WANT.

Also, there is a pdf that explains how to run the whole project but inside Openbox which it uses some prepared files for specific fpga, which couldn't help me at this moment.
many thanks for your explanation and advice. I hope I finally figure out to run one simple tsn switch in my board after a while :D

**Comment by hakiri** [comment]
Software is available in src/software code.
Use QT for Windows (or your preferred OS) to see all projects

**Comment by Adnan1986** [comment]
Yes I know, for this part(software), do you think I could find a  document about how to run this part and what files should be run?
